### DISCUSSION TITLE
How to contribute to this repo?

### DISCUSSION THREAD

**Discussion by alephpiece** [aidenlx/obsidian-zotero#240] [Help] [answered]
I tried to prepare a PR to make some minor fixes but failed to pull some dependencies required by `rush.json`. I don't understand the repo structure well - submodules like `ophidian-lib-core` seem to be private projects? Is there any way to contribute to this repo, or is a contribution acceptable?

**Comment by aidenlx** [comment]
Thank you for your interest in contributing to the project. I appreciate your effort to prepare a PR. However, I must inform you that the project is currently not ready for external contributions. This is primarily due to the complexity of the build system and the fact that I haven't had the time to write a detailed contribution guide.

At this stage, I'm focusing on streamlining the project and considering ways to make it more accessible for contributors. I'll keep the community updated on any progress in this regard. For now, I'd recommend holding off on contributions, but I truly value your willingness to contribute and hope to make this possible in the future.

**Reply by l1mey112** [reply]
How do I just build this project? I want to make a minor change. This repo is borderline proprietary.

**Reply by aidenlx** [reply, marked as answer]
Yeah, you're hitting the exact problem that's been bugging me! ZotLit was a hobby project with experimental architecture that, looking back now, was unnecessarily complex - even I get confused by my own code sometimes 😅
I'm currently doing a major refactor to simplify everything and make it actually possible for people to contribute. The complex build system and weird dependencies are exactly what I'm trying to fix.
If you're curious why I stepped away from maintenance and why I'm back to fixing this mess, I wrote about it here: [Reconnecting the Dots: A Journey from Medicine to Open Source](https://aidenlx.site/blog/reconnecting-dots)
Once the refactor is done, I'll definitely want community contributions and will make sure the setup actually works this time.
